Disable/enable ops initial implementation

milismsft opened the pull request:
Return proper create time and image name when querying server group details.
Add initial implementation for disable/enable ops; we will revisit these when moving to use AppGateway; for the time being we will power down/restart the instances only.
Other minor fixes.

@JargoonPard @scotmoor PTAL

Commit:
provider/kubernetes: Cache jobs and their pods

JargoonPard commented:
LGTM

scotmoor commented on clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureComputeClient.groovy:
```diff
@@ -174,6 +174,27 @@ public class AzureComputeClient extends AzureBaseClient {
+
+  ServiceResponse<Void> enableServerGroup(String resourceGroupName, String serverGroupName) {
+    VirtualMachineScaleSetsOperations ops = getAzureOps(
+      client.&getVirtualMachineScaleSetsOperations, "Get operations object", "Failed to get operation object") as VirtualMachineScaleSetsOperations
```
Why do we have retry logic around the "get operations" methods? These methods just instantiate and initialize the class that defines the available calls to Azure for that resource . There's no rest call actually being made. I thought the retry logic was for when we actually made a call to Azure

milismsft replied:
When I briefly looked at the JSDK implementation for getting the ops object and I could not guarantee that they don't make a rest call. For instance what if the current token/credentials are expired? Don't they need to renew them as part of this and isn't that another REST call? 

Can you please ask the JSDK guys to confirm that retrieving the operation object is 100% guaranteed not to make a REST call into Azure? If the answer is yes, I will remove that code.

lwander committed:
Merge pull request #513 from lwander/kubernetes-poll-jobs

provider/kubernetes: Cache jobs and their pods

cfieber committed:
dependency cleanup

for some reason frigga was added in cats-core but never used and not added in clouddriver-core...

turns findbugs back on with newer toolVersion

cfieber committed:
Merge pull request #525 from cfieber/cats_deps_findbugs_fix

dependency cleanup

milismsft committed:
Return proper create time and image name when querying server group details.
Add initial implementation for disable/enable ops; we will revisit these when moving to use AppGateway; for the time being we will power down/restart the instances only.
Other minor fixes.

Commit:
Remove unused DeleteAsg operation. (#526)

Commit:
provider/kubernetes: Set user agent

Commit:
provider/kubernetes: Enable specifying image w/o dropdown

lwander committed:
Merge pull request #528 from lwander/kubernetes-ua

provider/kubernetes: Set user agent

lwander committed:
Merge pull request #529 from lwander/kubernetes-imageid-pref

provider/kubernetes: Enable specifying image w/o dropdown

Commit:
provider/kubernetes: Include version in UA

Commit:
Kill off unused AWS ShrinkCluster operation. (#530)

lwander committed:
Merge pull request #531 from lwander/kubernetes-ua-2

provider/kubernetes: Include version in UA

Commit:
Add flag to build uber jar (#532)

* Add flag to build uber jar

Deploying clouddriver to Cloud Foundry requires an uber jar. This adds a flag "springBoot.repackage" that can be used on the command line to build an uber jar.

* provider/cloudfoundry: Handle credentials for a clean install
